src/main/java/org/fisco/bcos/web3j/crypto/gm/sm2/SM2Signer.java

```diff
@@ -0,0 +1,384 @@
+                }
+            }
+
+            digest.update(z, 0, z.length);
+        } else {
+            ecKey = (ECKeyParameters) baseParam;
+            ecParams = ecKey.getParameters();
+            pubPoint = ((ECPublicKeyParameters) ecKey).getQ();
+            z = getZ(userID);
+
+            digest.update(z, 0, z.length);
+        }
+    }
+
+    @Override
+    public void update(byte b) {
+        digest.update(b);
+    }
+
+    @Override
+    public void update(byte[] in, int off, int len) {
+        digest.update(in, off, len);
+    }
+
+    public void updateMessage(byte[] in, int off, int len) {
+        digest.reset();
+        digest.update(in, 0, in.length);
+        byte[] md = digestDoFinal();
+        digest.update(z, 0, z.length);
+        digest.update(md, 0, md.length);
+    }
+
+    @Override
+    public boolean verifySignature(byte[] signature) {
+        try {
+            BigInteger[] rs = derDecode(signature);
+            if (rs != null) {
+                return verifySignature(rs[0], rs[1]);
+            }
+        } catch (IOException e) {
+        }
+
+        return false;
+    }
+
+    @Override
+    public void reset() {
+        digest.reset();
+
+        if (z != null) {
+            digest.update(z, 0, z.length);
+        }
+    }
+
+    @Override
+    public byte[] generateSignature() throws CryptoException {
+        byte[] eHash = digestDoFinal();
+
+        BigInteger n = ecParams.getN();
+        BigInteger e = calculateE(eHash);
+        BigInteger d = ((ECPrivateKeyParameters) ecKey).getD();
+
+        BigInteger r, s;
+
+        ECMultiplier basePointMultiplier = createBasePointMultiplier();
+
+        // 5.2.1 Draft RFC:  SM2 Public Key Algorithms
+        do // generate s
+        {
+            BigInteger k;
+            do // generate r
+            {
+                // A3
+                k = kCalculator.nextK();
+
+                // A4
+                ECPoint p = basePointMultiplier.multiply(ecParams.getG(), k).normalize();
+
+                // A5
+                r = e.add(p.getAffineXCoord().toBigInteger()).mod(n);
+            } while (r.equals(ZERO) || r.add(k).equals(n));
+
+            // A6
+            BigInteger dPlus1ModN = d.add(ONE).modInverse(n);
+
+            s = k.subtract(r.multiply(d)).mod(n);
+            s = dPlus1ModN.multiply(s).mod(n);
+        } while (s.equals(ZERO));
+
+        // A7
+        try {
+            return derEncode(r, s);
+        } catch (IOException ex) {
+            throw new CryptoException("unable to encode signature: " + ex.getMessage(), ex);
+        }
+    }
+
+    public BigInteger[] generateSignature2() throws CryptoException {
+        byte[] eHash = digestDoFinal();
+
+        BigInteger n = ecParams.getN();
+        BigInteger e = calculateE(eHash);
+        BigInteger d = ((ECPrivateKeyParameters) ecKey).getD();
+
+        BigInteger r, s;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/fisco/web3sdk/pullRequest?prid=4981260)